WordPress/src/main/java/org/wordpress/android/ui/sitecreation/segments/NewSiteCreationSegmentViewHolder.kt

```diff
@@ -0,0 +1,39 @@
+package org.wordpress.android.ui.sitecreation.segments
+
+import android.support.annotation.LayoutRes
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.vertical.VerticalSegmentModel
+import org.wordpress.android.util.StringUtils
+import org.wordpress.android.util.image.ImageManager
+import org.wordpress.android.util.image.ImageType.IMAGE
+
+sealed class NewSiteCreationSegmentViewHolder(internal val parent: ViewGroup, @LayoutRes layout: Int) :
+        RecyclerView.ViewHolder(LayoutInflater.from(parent.context).inflate(layout, parent, false)) {
+    abstract fun onBind(model: VerticalSegmentModel, isLast: Boolean)
+
+    class SegmentViewHolder(
+        parentView: ViewGroup,
+        private val imageManager: ImageManager,
+        private val onItemTapped: (VerticalSegmentModel) -> Unit
+    ) : NewSiteCreationSegmentViewHolder(parentView, R.layout.new_site_creation_segment_item) {
+        private val container = itemView.findViewById<ViewGroup>(R.id.container)
+        private val icon = itemView.findViewById<ImageView>(R.id.icon)
+        private val title = itemView.findViewById<TextView>(R.id.title)
+        private val subtitle = itemView.findViewById<TextView>(R.id.subtitle)
+        private val divider = itemView.findViewById<View>(R.id.divider)
+
+        override fun onBind(model: VerticalSegmentModel, isLast: Boolean) {
+            title.text = model.title
+            subtitle.text = model.subtitle
+            imageManager.load(icon, IMAGE, StringUtils.notNullStr(model.iconUrl))
```

Comment:
```suggestion
            if(!model.iconUrl.isBlank()) {
                imageManager.load(icon, IMAGE, model.iconUrl)
            } else {
                imageManager.cancelRequestAndClearImageView(icon)
            }
```
Do we need to do something like this ^? The `ImageManager.load` method says that the empty values will be used to load the placeholder, but it doesn't actually cancel the previous request, does it? If it does, it kind of makes me question why we use the `cancelRequestAndClearImageView` in some cases that we do in other pages.

At the very least, I think we can get rid of the `StringUtils.notNullStr` since it's not a null-able value. I can see an argument for keeping it in case for future changes, but that's a slippery slope and I feel like if it's best to keep the code clean.

Reply:
Good question!
- AFAIK invoking a load method on the same ImageView automatically cancel the previous request
- We use `cancelRequestAndClearImageView` for cases where for example the view for that particular item is completely hidden - as we don't want to show any placeholder + we want to free the memory --->> we could theoretically call load with "", but that would make the code unreadable in my opinion


>At the very least, I think we can get rid of the StringUtils.notNullStr since it's not a null-able value. 
I agree -> I was developing it against the previous branch where the `iconUrl` was nullable.

Reply:
👍